### PR TITLE
[WIP] Download preload urls in the Preload constructor 

### PR DESCRIPTION
**Pull request by marcosmoyano**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed` / `isort distributed`

Follow up from: https://github.com/dask/distributed/pull/5185

Go back to processes and waiting a bit until the process and the webserver are up.  

I'll leave #5185 open for a bit and try to figure out what's the difference between the environments.

### CODE REVIEW

**Comment by mrocklin** [comment]
Let's try adding a longer timeout?

**Comment by mrocklin** [comment]
Nope.  Still a problem.

**Comment by marcosmoyano** [comment]
hhhmmm 😞  that's the only env where these tests are failing. Unfortunately I don't have much macos experience to know what's going on here. I'll keep looking later today.

**Comment by mrocklin** [comment]
We can also add a version dependent skip.  This is probably past the point
of diminishing returns.

**Comment by marcosmoyano** [comment]
That seemed to have done it. The failing tests don't look related to my changes.

File: distributed/preloading.py [hunk 1]
```diff
@@ -119,13 +119,14 @@ def _import_module(name, file_dir=None) -> ModuleType:
     return module
 
 
-async def _download_module(url: str) -> ModuleType:
+def _download_module(url: str) -> ModuleType:
```
Comment (jrbourbeau) [inline distributed/preloading.py:122]: Apologies for being late to the party. Could you elaborate a bit more on why this change was needed? My guess is there was a use case where we wanted the preload to be downloaded before starting up a worker or nanny. But I personally would find a little more explanation for this change useful

Comment (mrocklin) [reply distributed/preloading.py:122]: Preloads have actions that happen on ...

1.  Import
2. Setup
3. Teardown

*Except* if the preload is a web preload, in which case the import happens at start.  It's an odd exception that this PR removes.  

For the practical use case this has to do with workers asking another service what the scheduler address is when they start up.

Comment (marcosmoyano) [reply distributed/preloading.py:122]: Thanks Matt for the quick reply. I definitely need to do a better job adding context to my PRs ☹️. I apologize @jrbourbeau that's my fault.

**Comment by jrbourbeau** [comment]
This PR implicitly adds `urllib3` as a dependency to `distributed`. @marcosmoyano is that necessary, or just for convenience? I'm wondering if there's something in standard library we can use instead. 

xref [this CI build over in `dask/dask`](https://github.com/dask/dask/pull/8015/checks?check_run_id=3294211667) where this came up

**Comment by marcosmoyano** [comment]
> is that necessary, or just for convenience? I'm wondering if there's something in standard library, or one of our existing dependencies, we can use instead.

it was just for convenience. I'm sorry I thought urllib3 was already a dependency. I'll create a quick PR using `urllib` instead

**Comment by jrbourbeau** [comment]
That would be great, thank you @marcosmoyano!
